examples/todos/src/containers/VisibleTodoList.spec.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { mount, shallow } from 'enzyme'
+import { createStore } from 'redux'
+import { Provider } from 'react-redux'
+
+import rootReducer from '../reducers'
+import VisibleTodoList, { mapStateToProps } from './VisibleTodoList';
+import TodoList from '../Components/TodoList';
+import { VisibilityFilters } from '../actions'
+
+
+
+describe("integration test, high verification", () => {
+    it('renders an empty list', () => {
+        const store = createStore(rootReducer)
+        const wrapper = mount(
+            <Provider store={store}>
+                <VisibleTodoList />
+            </Provider>
+        )
+        expect(wrapper.html()).toBe('<ul></ul>')
+    })
+})
+
+describe("unit tests, high isolation", () => {
+    describe("mapStateToProps", () => {
+        it('maps an empty list to empty list', () => {
+            const props = mapStateToProps({
+                todos: [],
+                visibilityFilter: VisibilityFilters.SHOW_ALL
+            })
+            expect(props).toEqual({ todos: [] })
+        });
+    })
+
+    describe("<VisibleTodoList />", () => {
+        it('supplies an empty list to the <TodoList />', () => {
+            const store = createStore(rootReducer)
+            const wrapper = mount(
+                <Provider store={store}>
+                    <VisibleTodoList />
+                </Provider>
+            )
+            expect(wrapper.find('TodoList').prop('todos')).toEqual([])
```

Comment:
@kentcdodds While I'm well aware of the advantages & drawbacks of testing this way, suppose you wanted to write this test, how would you go about writing this sort of test in RTL? Would you not just end up re-implementing Enzyme's affordances in a round about way, rolling your own abstractions to polyfill what RTL intentionally doesn't implement?

Reply:
https://kcd.im/imp-deets

Reply:
I'm aware of the advantage & drawbacks, I'm not looking to debate those tradeoffs. I'm saying, assuming someone wants to make this tradeoff (which I'm aware you disagree with), RTL has no affordance for it like Enzyme does, right?

Reply:
Wha I would do is query the output for the todos that are rendered. The user doesn't care about the "TodoList" component or the "todos" props.

Reply:
I understand that's what you'd would do, and I wholeheartedly agree that it's preferable! 99.99% of the time!

Imagine a click handler that updates state, and then there's a rube goldberg machine of cascading updates. Its ideal to extract a pure function, and unit test that, but in extracting the pure function you may break the component. It can be ideal is have a unit test that asserts on implementation details, so you can in turn extract a pure function, and in turn replace the "bad" unit test with a "good" unit test of the pure function.  Having the option to work incremental can be helpful.

My point is, whether you agree people should write these tests or not, RTL doesn't have the option to use escape hatches (which can be desirable or undesirable depending on who you ask). I'm just trying to confirm that in this situation, you'd have to make a predetermined tradeoff in RTL whereas in Enzyme you'd have the option to change the tradeoffs.

I'm aware we disagree on the "sometimes its ok to cheat" part, I'm just trying to make sure I'm not off spouting factual inaccuracies about the pros & cons of the frameworks.